src/manager.rs

```diff
@@ -102,25 +102,28 @@ impl<S: EventSubscriber> EventManager<S> {
 
     /// Run the event loop blocking until the first batch of events.
     // Using the current run interface, not sure if the best but that's another discussion.
-    pub fn run(&mut self) -> Result<usize> {
+    pub fn run(&mut self) -> Result<()> {
```

Comment:
I'm not sure what we're gaining by removing this information.

This is actually useful in tests where you want to test event handling code for components using `EventManager`. There could be even production scenarios where this is useful.

Reply:
Looking at Firecracker code (including the tests) we just ignore it there as well. I guess the scenario for run is that you call it in a loop, and I don't see why you should care that it returned 0 events as it is not in the caller concern to do anything about it. Dispatching events is in the control of the event manager. Do you see this differently?

Reply:
There are Firecracker tests ([here](https://github.com/firecracker-microvm/firecracker/blob/ce7a3d9b7675c93296b3f62b143a67f964d4c90e/src/devices/src/virtio/block/event_handler.rs#L159) is one example, full search [here](https://github.com/firecracker-microvm/firecracker/search?q=run_with_timeout&unscoped_q=run_with_timeout)) which make use of this information.

> I don't see why you should care that it returned 0 events as it is not in the caller concern to do anything about it.

This is an assumption on the caller's program logic. Firecracker for example doesn't use that information, but I can imagine programs that might have logic based on the number of events handled.

My confusion comes from **why** this change is done. Sure we can remove it (refactor some tests that use it, maybe add it back later if someone really needs it), but what do we gain by removing it?

Reply:
I was cleaning up the interface while writing documentation for it. As `event_count` does not appear to be useful (subjective view), I want to remove it to make the interface simple. I am pretty much trying to not give information to the user that it's not needed for them to properly use the crate. Without a real use case scenario where this is used, I would remove it for now (before publishing the interface to crates.io).

@jiangliu @sboeuf do you need the event_count by any chance?

Reply:
Fine with me, and as you mentioned, if this is needed at some point, it can be added back, no big deal.

Reply:
Suggest to keep the return value, so we have the chance to tune IO thread priority of based on the IO events handled per loop.

Reply:
@jiangliu I'll keep it then, thanks everyone.